ethereum/contracts/relayer/coreRelayer/CoreRelayerState.sol

```diff
@@ -15,9 +15,8 @@ contract CoreRelayerStorage {
     struct State {
         Provider provider;
         // delivery lock for reentrancy protection
-        bool contractLock;
```

Comment:
This is duplicate state that needs to be kept in sync. By using the zero-address  as "unlocked" we prevent invalid state by construction

Reply:
One thing to remember with this is we are also using 'targetAddress 0 address' for cross chain refunds. This means that our decision to skip the main codepath for these cross chain refunds (and just pay the refund and return immediately) is not just an efficiency thing now, but is actually required for security reasons. I think this is fine, but yeah just want to point this out.

Reply:
yeah I'm actually going to roll back this change for this reason (and it changes the storage format...)